cpu/cortexm_common/Makefile.features

```diff
@@ -5,3 +5,7 @@ FEATURES_PROVIDED += periph_pm
 FEATURES_PROVIDED += cpp
 FEATURES_PROVIDED += cpu_check_address
 FEATURES_PROVIDED += ssp
+
+ifneq ($(CPU_ARCH),cortex-m0)
+  FEATURES_PROVIDED += cortexm_mpu
+endif
```

Comment:
According to [Wikipedia](https://en.wikipedia.org/wiki/ARM_Cortex-M#Silicon_customization), the MPU is always optional.

Reply:
Hmok. Do all our non-cm0 CPUs have an MCU?
I'll add a check to the mpu test application.

Reply:
I've added a check. Seems like all our non-Cortex-M0 boards have an MPU, or at leastm `__MPU_PRESENT` set.

Reply:
Either `__MPU_PRESENT` is not undefined, or *all* our Cortex-M boards *do* have the optional MPU, even the CM0. @aabadie Could you verify the test succeeds e.g., on a stm32f031k6?

Reply:
@aabadie could you take a look at this? I don't have a M0 board...

Reply:
I tried on nucleo-f072rb, which is an M0, and the `tests/mpu_stack_guard` is correctly skipped by the build system.

The problem here is that, according to the __MPU_PRESENT macro in vendor files, stm32f1 and also some stm32f3 (so not all) CPUs  doesn't provide an MPU. It's also unclear if stm32l0 provides an MPU or not (I'm wondering if the vendor files were manually altered for this family).

So, as suggested by @benpicco, I think this feature should be provided by default less widely. Maybe just for M4/M7 by default, and enable on a case-by-case basis for other CPU families (M0/M0plus,M3).

Reply:
I just checked the data sheets, neither samd2x nor saml2x (cm0p) implement an MPU.